core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -296,22 +296,25 @@ public long getDefaultBlockSize() {
         // add the existing in-Alluxio block locations
         List<WorkerNetAddress> locations = fileBlockInfo.getBlockInfo().getLocations()
             .stream().map(alluxio.wire.BlockLocation::getWorkerAddress).collect(toList());
-        if (locations.isEmpty()) {
-          // No in-Alluxio location, fallback to use under file system locations with
-          // co-located workers.
-          if (workerHosts == null) {
+        if (locations.isEmpty()) { // No in-Alluxio location
+          if (workerHosts == null && (!fileBlockInfo.getUfsLocations().isEmpty() || Configuration
```

Comment:
Should this "initialization" just happen in each case block ( with `if (workerHosts == null)` )? It might be easier to follow? This if-condition is basically duplicating the conditions of the 2 cases below.